webpack-build-utils/presets/webpack.hardsource.js

```diff
@@ -0,0 +1,6 @@
+const HardSourceWebpackPlugin = require('hard-source-webpack-plugin');
+
+// for more on this plugin see https://github.com/mzgoddard/hard-source-webpack-plugin
+module.exports = () => ({
+    plugins: [new HardSourceWebpackPlugin()]
```

Comment:
Out of curiosity, why not put this in `webpack.development.js`? Or to take this to an extreme: do we want separate files for each plugin?

Reply:
It is a nice way to compose things into different script targets/environments. Also, this idea is something that will be coming to Webpack natively soonish https://gist.github.com/TheLarkInn/684bcdb5cf3a082c5a3cee907168c59a

It also makes it super simple to remove a specific plugin either entirely or only from some environment targets. It keeps the configs cleaner which is a blessing in itself ;p Once we know something is going to stick around for the long haul, we might decide to move it directly into the main dev config.

Reply:
Cool, I like that two stage model 👍 Thanks for the elaboration!